### ISSUE TITLE
SSIM Loss Layer

### ISSUE THREAD

**Issue by DanielBonanno** [mehtadushy/DeepNetStuff#2]
Dear Mehta,
Is the SSIM implementation correct for the GPU or is it still without the GPU implementation? 
Plus, does it work if the weight is left as 1 or does it need to be scaled so that the network can be trained? Currently im trying it on 64x64 patches and im getting an SSIM of approximately 64, which should not be the case.

Also, I'm very new to caffe, so please excuse this simple question: Can this layer be used in conjunction with your L1 Loss layer?

Thank you.

**Comment by mehtadushy** [comment]
Yes, can be used in conjunction with the L1 loss layer. 

You'll have to work out the loss weight yourself though, based on the problem you are using it for, your network, learning rate, solver, other losses .......

It is a DSSIM layer, and is not normalized by the number of pixels, so scales with it. The [forward pass](https://github.com/mehtadushy/DeepNetStuff/blob/master/caffe/src/caffe/layers/ssim_loss_layer.cpp#L108) will tell you what is exactly happening in the code
